File: srv/src/main/java/customer/ai2code/model/Bot.java
package customer.ai2code.model;

import java.util.concurrent.Executor;

import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;

import cds.gen.mainservice.BotInstancesExecuteContext;

public interface Bot {
    public final Executor executor = null;

    public BotInstancesExecuteContext.ReturnType execute();

    public Boolean executeAsync();

    public Boolean stop();

    public Boolean resume();

    public Boolean cancel();

    public SseEmitter chatInStreaming(String content);

    public String chat(String content);
}

File: srv/model/Bot.ts
import { executeresult } from "#cds-models/";
import { BotInstances } from "#cds-models/MainService";

import { AIModel } from "./AIModel";

export interface Bot {
  execute(): executeresult;

  executeAsync(): boolean;

  stop(): boolean;

  resume(): boolean;

  cancel(): boolean;

  // chatInStreaming(content: string): SseEmitter;

  chat(content: string): string;

  getBotInstance(): BotInstances;

  getAiModel(): AIModel;
}

